### ISSUE TITLE
Better way to encourage users to add categories, depictions

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#3685]
When uploading an image using the app we seem to be showing a dialog box when a user selects no categories. Further, in the `structured-data` branch we're adding another step to the upload flow which asks users to add depictions of the images. In that step too, we're planning on showing a dialog when a user selects no depictions.

To me, these dialogs that we show when a user tries to skip a step is like trying to extract as much information from the uploader itself as possible. While that's a good ideally, this seems to go against the main advantages of a [_wiki_](https://en.wikipedia.org/wiki/Wiki) which is _collaborative modification_ and the fact that _anyone can edit it at anytime_. We aren't mandating the steps yet but just showing the dialog _prompt_ might give the opinion that the user is making the image less valuable by not adding depictions/categories _when they upload it_. The fact that we don't even have a "Don't show this again" dialog just worsens this. This should not be the case. They can actually come again later and improve the image if they wish (Psst, a possible enhancement which doesn't seem to have been filed yet) (or) they could just leave it to the others to do that stuff.

To be clear, I'm not against encouraging users to add categories/depictions. I'm just saying dialogs aren't the best way to do that. Possible alternatives to dialogs:
* In the free space found in the bottom of the respective step educate the users about how adding categories, depictions would help with images being used widely.
* Show a "Possible ways for images to be better used" section in the last page of the upload screen suggesting that the user can add categories, depictions if they haven't chosen any during the steps.
If we get this _subtler encouragement_ right, an uploader who really wants their images to be identified via categories and depictions would add the categories, depictions. If the uploader thinks that the title, description is enough just allow them to exercise their freedom and go with their flow.

Kindly share your thoughts about this.

**Comment by nicolas-raoul** [comment]
It is true that the web Upload Wizard does not strictly require one, but https://commons.wikimedia.org/wiki/Commons:Categories says:

> It is essential that every file can be found by browsing the category structure. To allow this, each file must be put into a category directly.

"Must" is a strong word, so I believe our warning is not too much.
How about waiving it for high-level users, in the spirit of https://github.com/commons-app/apps-android-commons/issues/2852?

> free space found in the bottom of the respective step

Actually when suggestions are implemented, there will almost never be any free space. But how about putting this explanation inside the warning popup?

Adding an extra step before upload for media that could be made more discoverable is an interesting idea. But I think the current popups are easier to understand by all users, as they come just after the user ignored a selection activity :-)

**Comment by misaochan** [comment]
Yeah, it is tough to find an ideal balance between how much or how little we should nudge users to do certain things. FWIW, when I proposed the warning dialogs many years back and sought input from the community, some of the input I received was that the "warnings are good but not enough". ;) 

> In the free space found in the bottom of the respective step educate the users about how adding categories, depictions would help with images being used widely.

This is a good idea, however I feel it should be in addition to the warnings, not replacing them.

> Show a "Possible ways for images to be better used" section in the last page of the upload screen suggesting that the user can add categories, depictions if they haven't chosen any during the steps.

This could be a possible replacement, but how do you feel it would be beneficial to users compared to the dialogs?

**Comment by macgills** [comment]
I am in favour of just keeping the dialogs

**Comment by sivaraam** [comment]
>> It is essential that every file can be found by browsing the category structure. To allow this, each file must be put into a category directly.
> 
> "Must" is a strong word, so I believe our warning is not too much.

Let me repeat myself:

> To be clear, I'm not against encouraging users to add categories/depictions. I'm just saying dialogs aren't the best way to do that.

So, I'm not against encouraging users to add categories and depictions at all.

> But I think the current popups are easier to understand by all users, as they come just after the user ignored a selection activity :-)

It's not that they are not understandable. It's that some might feel it annoying that we keep on showing suggestion dialogs for something they might wish to ignore (see also: #3581). Also, it's not like a user can categorize or add depictions only when they upload. They actually can do that even after they have uploaded it. But I won't be surprised if our dialogs mislead someone into thinking that they can't categorize or add depictions after they have uploaded the image. We can avoid this misleading assumption if we communicate it better. BTW, editing the image information such as description, categories after upload would be a useful enhancement to the app.

>> In the free space found in the bottom of the respective step educate the users about how adding categories, depictions would help with images being used widely.
> 
> This is a good idea IMO (if it works with the rest of the UI), however I feel it should be in addition to the warnings, not replacing them.

If we do that in addition to the dialog, I feel it would come out as annoying as we would already have communicated that images with categories are discovered better. That's why I suggested it as a replacement.

>> Show a "Possible ways for images to be better used" section in the last page of the upload screen suggesting that the user can add categories, depictions if they haven't chosen any during the steps.
>
> This could be a possible replacement, but how do you feel it would be beneficial to users compared to the dialogs?

TL;DR: fewer clicks

If we show a separate final step with suggestions, I believe it would be better as we can show all suggestions/warnings in one place. This might also solve the [problem with validation flow](https://github.com/commons-app/apps-android-commons/issues/3500#issuecomment-601157407). I'm thinking of something like this:

![20200426_232710-1-cropped](https://user-images.githubusercontent.com/12448084/80315779-6a768900-8817-11ea-909a-d31f8ac40cad.jpg)

I hope that gives an idea. Apologies if the mockup doesn't look so good, though :)

**Comment by misaochan** [comment]
Interesting! Yes, I think that if validations are overhauled in a similar manner, providing the category/depiction warnings in the validations page could be a good idea. However this would depend on the validations being overhauled first.

**Comment by nicolas-raoul** [comment]
Does that mean that when Joe selects a picture from the Internet for upload, we don't warn him immediately, and instead let him spend minutes filling metadata before warning him? That would make him much more likely to ignore the warning, because he has spent effort writing the metadata. Better teach people as soon as we can.

**Comment by misaochan** [comment]
Hmm, good point, postponing validation until after depictions and categories are selected could be problematic in that aspect. :/ I had forgotten that even if we were to overhaul validations, they should be displayed before the depicts step.
